pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1487,6 +1498,41 @@ static void updateColumnNames(String rawTableName, PinotQuery pinotQuery, boolea
     }
   }
 
+  private static void expandStarExpressionToActualColumns(PinotQuery pinotQuery, Map<String, String> columnNameMap,
+      Expression selectStarExpr) {
+    List<Expression> originalSelections = pinotQuery.getSelectList();
+    Set<String> originallySelectedColumnNames =
```

Comment:
Avoid using stream apis in query path because we have found that it has poorer performance comparing to regular apis.

Reply:
Done, thanks. I was unaware of this. Is there a benchmark that I can refer to for my own knowledge? There's literature available online, but having Pinot specific info would be handy. Thanks!